compiler/lexer.nim

```diff
@@ -340,6 +340,21 @@ proc getNumber(L: var Lexer, result: var Token) =
     L.bufpos = msgPos
     lexMessage(L, msgKind, msg % t.literal)
 
+  proc cmpSuffix(L: Lexer, prev: int, endOfSuffix: int, chs: array[1, char]): bool =
+    if (endOfSuffix - prev) != 2:
+      return false
+    return L.buf[prev + 1] == chs[0]
+
+  proc cmpSuffix(L: Lexer, prev: int, endOfSuffix: int, chs: array[2, char]): bool =
+    if (endOfSuffix - prev) != 3:
+      return false
+    return L.buf[prev + 1] == chs[0] and L.buf[prev + 2] == chs[1]
+
+  proc cmpSuffix(L: Lexer, prev: int, endOfSuffix: int, chs: array[3, char]): bool =
+    if (endOfSuffix - prev) != 4:
+      return false
+    return L.buf[prev + 1] == chs[0] and L.buf[prev + 2] == chs[1] and L.buf[prev + 3] == chs[2]
```

Comment:
```suggestion
  proc cmpSuffix(L: Lexer, prev: int, endOfSuffix: int, chs: openArray[char]): bool =
    if (endOfSuffix - prev - 1) != chs.len:
      return false
    for i in 0..<chs.len:
      if L.buf[prev + 1 + i] != chs[i]: return false
    return true
```

Reply:
Yours is cleaner but I suspect mine is faster as it avoids a loop for speculative cpu execution.

But I suspect clarity is more important, so I'll change it to yours.

Reply:
I'd be surprised if a C compiler with optimization turned on doesn't unroll the loop. And in any case, this isn't a hotspot.

(please also add {.inline.} although in this case it probably makes no difference given that inlining is possible within same module)